route.go

```diff
@@ -322,6 +322,10 @@ func (m methodMatcher) Match(r *http.Request, match *RouteMatch) bool {
 // Methods adds a matcher for HTTP methods.
 // It accepts a sequence of one or more methods to be matched, e.g.:
 // "GET", "POST", "PUT".
+//
+// NOTE: Chaining multiple Methods will result in the last added
+// Methods will override the allowed methods specified by the
```

Comment:
I don't think this is not the actual behaviour, have you tried this?  when I write something like the below:
```golang
r := mux.NewRouter()
r.HandleFunc("/health", healthHandle).Methods("POST").Methods("GET")
```

When I curl `/health` endpoint, it returns `HTTP ERROR 405`

Reply:
@amustaque97 this is weird but I am getting `405` for both the `POST` and `GET` request :thinking:

Reply:
@henrikac  this is expected behaviour. Consider it as series of filter. When you filter the requests based on POST and then try to do it with GET, the result will be 0 and hence 405 error

Reply:
> @henrikac this is expected behaviour. Consider it as series of filter. When you filter the requests based on POST and then try to do it with GET, the result will be 0 and hence 405 error

Yeah, I figured. In my comment below for the README I mention that I think it looks like it behaves like it performs an intersection operation :)

Reply:
@henrikac so is this PR required? Could you modify or decline PR based on changes/feedback